caluma/form/serializers.py

```diff
@@ -563,6 +532,17 @@ class SaveDocumentTableAnswerSerializer(SaveAnswerSerializer):
         help_text="List of document IDs representing the rows in the table.",
     )
 
+    def _get_document_tree(self, document_id):
```

Comment:
Since you're not returning a nested structure here, you could just find all documents with the same family and return the bunch, doing one single query instead of recursively descending down the tables.

Also, as you're not really returning a tree, I'd suggest a different name (`_get_family_documents` comes to mind..)

Reply:
Related to #524